src/component/axis/AxisBuilder.ts

```diff
@@ -435,7 +435,7 @@ const builders: Record<'axisLine' | 'axisTickLabel' | 'axisName', AxisElementsBu
                 text: name,
                 font: textFont,
                 overflow: 'truncate',
-                width: maxWidth,
+                width: maxWidth || +textStyleModel.get('width'),
```

Comment:
Using `||` is not correct because `maxWidth` can be 0. You should update the logic at https://github.com/apache/echarts/pull/16206/files#diff-180b57aa9331c3128a2e4f577028b529d0318bf3eecc911499242546bf1012ceR425 in retrieve.

Reply:
Please also test @weixingqi 's report. Thanks!